src/tdl/queue/queue_based_implementation_runner.py

```diff
@@ -114,11 +115,13 @@ def process_next_request_from(self, remote_broker, headers, request):
         response = self._processing_rules.get_response_for(request)
         self._audit.log(response)
 
-        client_action = response.client_action
-        self._audit.log(client_action)
+        #TODO: check again if this is correctly done, come back later to complete it
```

Comment:
@julianghionoiu In python how do we want to handle these edge cases, if there are multiple messages and one of the middle ones fails due to any of the two reasons, we should terminate the message listening process

Reply:
At the moment we are not handling it effectively as you can see from the below code

Reply:
We still have a case of both the messages reaching the response queue

Reply:
If 3 messages are sent and then 2nd one fails during processing, so recevied messages is still 2 right? And published messaged is 1?

Reply:
Yes, that is correct.

You can remove the comment